BlueprintUI/Sources/Internal/SPL/SPCacheTree.swift

```diff
@@ -55,38 +55,136 @@ final class SPCacheTree<Key, Value, SubcacheKey> where Key: Hashable, SubcacheKe
 
 }
 
-extension SPCacheTree where Key == SizeConstraint {
+extension SPCacheTree where Key == SizeConstraint, Value == CGSize {
     func get(key: Key, or create: (Key) -> Value) -> Value {
         valueCache.get(key: key, or: create)
     }
 }
 
 final class SPValueCache<Key: Hashable, Value> {
     
+    typealias Options = SPCacheOptions
+
     var values: [Key: Value] = [:]
     
     var path: String
+    var options: Options
     
-    init(path: String) {
+    init(path: String, options: Options) {
         self.path = path
+        self.options = options
     }
 }
 
-extension SPValueCache where Key == SizeConstraint {
+struct SPCacheOptions {
+    var hintRangeBoundaries: Bool
+    var searchUnconstrainedKeys: Bool
+}
+
+extension SPValueCache where Key == SizeConstraint, Value == CGSize {
 
     func get(key: Key, or create: (Key) -> Value) -> Value {
         if let size = values[key] {
             Logger.logCacheHit(object: self, description: path, constraint: key)
             return size
         }
         
+        // TODO:
+        // This has a negative perf impact on deep stack tests. Need to determine what the impact
+        // is on a wider variety of tests, and if there is a way we can avoid this for cases
+        // where it doesn't help (e.g. not in a scroll view).
```

Comment:
Does this TODO need to be updated? It looks like we addressed this by introducing an option that's only true during layout.

Reply:
Er, hm, maybe we do still want to think through scoping this further, I'm not entirely sure.

Reply:
It's not yet addressed for the condition of "are we in a scroll view or other layout that measures unconstrained", which could be important.